### ISSUE TITLE
Duplicate React keys when using immediate mode

### ISSUE THREAD

**Issue by jperasmus** [AndreasFaust/gatsby-plugin-transitions#13]
Hi there,

I looked and didn't see anyone else with the same issue, so I'm not sure if this is a bug with the plugin or with my Gatsby setup. I recently installed this plugin and set it up as described in the README. Everything worked fine with the default `successive` mode, but as soon as I change it to `immediate` mode, I get a React warning/error that there are duplicate keys in the `TransitionViews` component.

```
index.js:2177 Warning: Encountered two children with the same key, `initial`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
    in div (created by TransitionViews)
    in TransitionViews (created by Layout)
    in TransitionProvider (created by Layout)
    in div (created by ForwardRef(Box))
    in ForwardRef(Box)
    in ForwardRef(Box) (created by Layout)
    in SystemProvider (created by Provider)
```

I took a quick look and from what I can see, in `src/transitions/views.js` the `views` array has duplicate entries (ie. `view.props.location.key` is the same). From what I can tell, the [following `useEffect` hook](https://github.com/AndreasFaust/gatsby-plugin-transitions/blob/d85b5eb16003bb537119f7f56df885970ac8e3a9/src/transitions/views.js#L33) gets called twice on page load: once where `currentLocation.key` is `undefined` and then with the proper values. Wrapping the body of the `useEffect` hook function to only add the view if the `currentLocation.key` is set seems to fix the problem. Would you be open to a PR if you consider this a bug with the plugin?

**Comment by AndreasFaust** [comment]
Thank you for figuring this out! 
I could not reproduce the error, but I did as you proposed and released version 2.1.6.
If the error persists, just reopen this issue.

**Comment by jperasmus** [comment]
Thanks for the quick fix. I found that the issue still sporadically occurred on my local (I think it has something to do with Gatsby's hot reloading while developing). What I also had to do was in the reducer.js file make sure that there are no duplicate routes/views. I basically just extended the existing `filterViews` helper to this:

```
function filterViews(views, currentView) {
  return views.filter(
    view =>
      view &&
      (!currentView ||
        view.props.location.key !== currentView.props.location.key)
  )
}
```

and in the 2 cases that uses it, I did this:

```
case 'ADD_VIEW_FROM_QUEUE':
      return {
        ...state,
        views: [state.queue, ...filterViews(state.views, state.queue)],
        queue: null
      }
case 'ADD_VIEW_DIRECTLY':
      return {
        ...state,
        views: [action.view, ...filterViews(state.views, action.view)],
        queue: null
      }
```

I haven't seen the issue since I made this change.
